apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/Config.java

```diff
@@ -148,6 +148,31 @@
          * If true, skywalking agent will enable profile when user create a new profile task. Otherwise disable profile.
          */
         public static boolean ACTIVE = true;
+
+        /**
+         * Parallel monitor segment thread count
+         */
+        public static int PARALLELS_THREAD_COUNT = 5;
+
+        /**
+         * Max monitor segment time(minutes), if current segment monitor time out of limit, then stop it.
+         */
+        public static int MAX_MONITOR_TIME = 10;
+
+        /**
+         * Max dump thread stack depth
+         */
+        public static int MAX_DUMP_STACK_DEPTH = 500;
+
+        /**
+         * Snapshot send to backend channel size
+         */
+        public static int SNAPSHOT_SEND_CHANNEL_SIZE = 2;
+
+        /**
+         * Snapshot send to backend buffer size
+         */
+        public static int SNAPSHOT_SEND_BUFFER_SIZE = 50;
```

Comment:
Same about the name. And is 50 enough? From my understanding, we do thread dump every 10ms, then we could have 100 in one second. I think as we could collect 5 parallel, you will highly trigger queue abandon mode easily. 
I suggest at least 5 * 500. @kezhenxu94 What do you think?

Reply:
Change to only one thread to process the profile task. So don't use `DataCarrier` anymore. Change to using `LinkedBlockingQueue` such as sniffer `JVMService`. using buffer size 500.